analyzers/src/SonarAnalyzer.VisualBasic/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -81,6 +81,9 @@ static bool TakesExpressionTree(SymbolInfo info)
             }
         }
 
+        public static bool IsShared(this SyntaxNode node) =>
```

Comment:
You will need a new class and the proper extension to check if the sub or function is shared:

```
internal static class MethodBlockSyntaxExtensions
{
    public static bool IsShared(this MethodBlockSyntax methodBlock) =>
        methodBlock.SubOrFunctionStatement.Modifiers.Any(SyntaxKind.SharedKeyword);
}
```